### PR TITLE
updated sample

### PR DESCRIPTION
**Pull request by salman90**
# Pull Request

## Purpose

* Update the following:

1. Readme.
2. PowerShell scripts.
3. Update logout function.
4. Msal-browser.

Thanks for your time.

## Does this introduce a breaking change

```console
    [ x ] Yes
    [ ] No
```

## Pull request type

What kind of change does this Pull Request introduce?

```console
    [  x ] Bugfix
    [ ] Feature
    [ x ] Code style update (formatting, local variables)
    [ x ] Documentation content changes
    [ ] Other... Please describe:
```

## How to test

* Get the code

```console
    git clone [repo-address]
    cd [repo-name]
    git checkout [branch-name]
    npm install
```

* Test the code

```console

```

## What to check

ex: verify that the following are valid:

* ...

### CODE REVIEW

File: 1-Authentication/1-sign-in/App/authConfig.js [hunk 1]
```diff
@@ -5,40 +5,39 @@
...
-            console.warn(message);
-            return;
-        }
-      }
-    }
-  }
+    auth: {
+        clientId: 'Enter_the_Application_Id_Here', // This is the ONLY mandatory field that you need to supply.
+        authority: 'https://login.microsoftonline.com/Enter_the_Tenant_Info_Here', // Defaults to "https://login.microsoftonline.com/common"
+        redirectUri: 'http://localhost:3000', // You must register this URI on Azure Portal/App Registration. Defaults to window.location.href
```
Comment (derisen) [inline 1-Authentication/1-sign-in/App/authConfig.js:11]:
```suggestion
        redirectUri: '/', // You must register this URI on Azure Portal/App Registration. Defaults to window.location.href e.g. http://localhost:3000
```

File: 1-Authentication/1-sign-in/App/authConfig.js [hunk 2]
```diff
@@ -5,40 +5,39 @@
...
+        navigateToLoginRequestUrl: true, // If "true", will navigate back to the original request location before processing the auth code response.
+    },
+    cache: {
+        cacheLocation: 'sessionStorage', // Configures cache location. "sessionStorage" is more secure, but "localStorage" gives you SSO.
+        storeAuthStateInCookie: false, // If you wish to store cache items in cookies as well as browser cache, set this to "true".
```
Comment (derisen) [inline 1-Authentication/1-sign-in/App/authConfig.js:16]:
```suggestion
        storeAuthStateInCookie: false, // set this to true if you have to support IE
```

File: 1-Authentication/1-sign-in/App/authPopup.js [hunk 3]
```diff
@@ -84,10 +84,12 @@ function signOut() {
 
     // Choose which account to logout from by passing a username.
     const logoutRequest = {
-        account: myMSALObj.getAccountByUsername(username)
+        account: myMSALObj.getAccountByUsername(username),
+        mainWindowRedirectUri: 'http://localhost:3000/signout',
+        redirectUri: 'http://localhost:3000/redirect.html',
```
Comment (derisen) [inline 1-Authentication/1-sign-in/App/authPopup.js:89]:
```suggestion
```

Comment (derisen) [inline 1-Authentication/1-sign-in/App/authPopup.js:89]:
```suggestion
        redirectUri: '/redirect', // e.g. http://localhost:3000/redirect
```

Comment (derisen) [reply 1-Authentication/1-sign-in/App/authPopup.js:89]: Here and everwhere we can get rid of `.html` extension in redirect uri (we had to do this in React to make it work with react router)

File: 1-Authentication/1-sign-in/README.md [hunk 4]
```diff
@@ -1,18 +1,35 @@
...
+languages:
+ - javascript
+products:
+ - azure-active-directory
+ - msal-js
+ - msal-browser
+urlFragment: https://github.com/Azure-Samples/ms-identity-javascript-tutorial
+extensions:
+- services: ms-identity
+- platform: DotNet
```
Comment (derisen) [inline 1-Authentication/1-sign-in/README.md:14]:
```suggestion
- platform: JavaScript
```

File: 1-Authentication/1-sign-in/README.md [hunk 5]
```diff
@@ -27,115 +44,115 @@ This sample demonstrates a Vanilla JavaScript single-page application (SPA) that
...
-#### Configure the app to use your app registration
+    1. In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `ms-identity-javascript-c1s1`.
+    1. Under **Supported account types**, select **Accounts in this organizational directory only**
+    1. Select **Register** to create the application.
+1. In the **Overview** blade, find and note the **Application (client) ID**. You use this value in your app's configuration file(s) later in your code.
+1. In the app's registration screen, select the **Authentication** blade to the left.
+1. If you don't have a platform added, select **Add a platform** and select the **Single-page application** option.
+    1. In the **Redirect URI** section enter the following redirect URIs:
+        1. `http://localhost:3000`
+        1. `http://localhost:3000/redirect.html`
```
Comment (derisen) [inline 1-Authentication/1-sign-in/README.md:139]:
```suggestion
        1. `http://localhost:3000/redirect`
```

File: 1-Authentication/1-sign-in/README.md [hunk 6]
```diff
@@ -250,46 +279,37 @@ For instance, to configure this sample for **Azure AD Germany** national cloud:
...
+* [Register an application with the Microsoft identity platform](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app)
+* [Configure a client application to access web APIs](https://docs.microsoft.com/azure/active-directory/develop/quickstart-configure-app-access-web-apis)
+* [Understanding Azure AD application consent experiences](https://docs.microsoft.com/azure/active-directory/develop/application-consent-experience)
+* [Understand user and admin consent](https://docs.microsoft.com/azure/active-directory/develop/howto-convert-app-to-be-multi-tenant#understand-user-and-admin-consent)
+* [Application and service principal objects in Azure Active Directory](https://docs.microsoft.com/azure/active-directory/develop/app-objects-and-service-principals)
+* [Authentication Scenarios for Azure AD](https://docs.microsoft.com/azure/active-directory/develop/authentication-flows-app-scenarios)
+* [Building Zero Trust ready apps](https://aka.ms/ztdevsession)
+* [National Clouds](https://docs.microsoft.com/azure/active-directory/develop/authentication-national-cloud#app-registration-endpoints)
+
```
[the remark this answers is not in the document]
Comment (derisen) [reply 1-Authentication/1-sign-in/README.md:308]: whitespace

File: 1-Authentication/1-sign-in/server.js [hunk 7]
```diff
@@ -2,11 +2,34 @@ const express = require('express');
 const morgan = require('morgan');
 const path = require('path');
 
+const rateLimit = require('express-rate-limit');
+
```
[the remark this answers is not in the document]
Comment (derisen) [reply 1-Authentication/1-sign-in/server.js:6]: whitespace

File: 1-Authentication/1-sign-in/README.md [hunk 8]
```diff
@@ -1,18 +1,35 @@
...
- 1. [Setup](#setup)
- 1. [Registration](#registration)
- 1. [Running the sample](#running-the-sample)
- 1. [Explore the sample](#explore-the-sample)
- 1. [About the code](#about-the-code)
- 1. [More information](#more-information)
- 1. [Community Help and Support](#community-help-and-support)
- 1. [Contributing](#contributing)
- 1. [Code of Conduct](#code-of-conduct)
+---
```
Comment (derisen) [inline 1-Authentication/1-sign-in/README.md:1]: @salman90 need to update README-incremental.md as well

File: 1-Authentication/1-sign-in/App/authConfig.js [hunk 9]
```diff
@@ -5,40 +5,39 @@
...
-            return;
-        }
-      }
-    }
-  }
+    auth: {
+        clientId: 'Enter_the_Application_Id_Here', // This is the ONLY mandatory field that you need to supply.
+        authority: 'https://login.microsoftonline.com/Enter_the_Tenant_Info_Here', // Defaults to "https://login.microsoftonline.com/common"
+        redirectUri: 'http://localhost:3000', // You must register this URI on Azure Portal/App Registration. Defaults to window.location.href
+        navigateToLoginRequestUrl: true, // If "true", will navigate back to the original request location before processing the auth code response.
```
Comment (derisen) [inline 1-Authentication/1-sign-in/App/authConfig.js:12]: what happened to the `postLogoutRedirectUri `config ?

File: 1-Authentication/1-sign-in/AppCreationScripts/sample.json [hunk 10]
```diff
@@ -1,29 +1,28 @@
...
-          "Comment": "Navigate to the Manifest page, find the entry for 'replyUrlsWithType' and change the value to 'Spa'"
-        }
-      ]
+        "Id": "client",
+        "Name": "ms-identity-javascript-c1s1",
+        "Kind": "SinglePageApplication",
+        "Audience": "AzureADMyOrg",
+        "HomePage": "http://localhost:3000",
+        "SampleSubPath": "1-Authorization\\1-sign-in",
+        "ReplyUrls": "http://localhost:3000, http://localhost:3000/redirect.html"
```
Comment (derisen) [inline 1-Authentication/1-sign-in/AppCreationScripts/sample.json:20]:
```suggestion
        "ReplyUrls": "http://localhost:3000, http://localhost:3000/redirect
```

**Comment by derisen** [comment]
@salman90 can we also add account switcher to all sign-in samples?

File: 1-Authentication/1-sign-in/AppCreationScripts/sample.json [hunk 11]
```diff
@@ -1,29 +1,28 @@
...
-      "ReplyUrls": "http://localhost:3000",
-      "ManualSteps": [
-        {
-          "Comment": "Navigate to the Manifest page, find the entry for 'replyUrlsWithType' and change the value to 'Spa'"
-        }
-      ]
+        "Id": "client",
+        "Name": "ms-identity-javascript-c1s1",
+        "Kind": "SinglePageApplication",
+        "Audience": "AzureADMyOrg",
```
Comment (kalyankrishna1) [inline 1-Authentication/1-sign-in/AppCreationScripts/sample.json:18]: >Audience [](http://example.com/codeflow?start=9&length=8)

Add 'acct' optional claim

**Review by kalyankrishna1 — APPROVED** [review]
:shipit:

File: 1-Authentication/1-sign-in/App/authConfig.js (hunk 9 above)
Comment (salman90) [reply 1-Authentication/1-sign-in/App/authConfig.js:12]: It is included in the redirect and popup request.

File: 1-Authentication/1-sign-in/server.js [hunk 12]
```diff
@@ -18,6 +40,10 @@ app.get('/signout', (req, res) => {
     res.sendFile(path.join(__dirname + '/App/signout.html'));
 });
 
+app.get('/redirect', (req, res) => {
+    res.sendFile(path.join(__dirname + '/public/redirect.html'));
```
Comment (derisen) [inline 1-Authentication/1-sign-in/server.js:44]: do we need to put `redirect.html` in public folder?

**Comment by derisen** [comment]
@salman90 could you also fix the failing CI in this PR?

**Comment by salman90** [comment]
> @salman90 could you also fix the failing CI in this PR?

Done!

**Pull request merged**